packages/frontend/amp/components/lib/GetImage.test.ts

```diff
@@ -0,0 +1,49 @@
+import { getImage } from './GetImage';
+
+test('chooses smallest image that is still greater than column width', async () => {
+    const images = [
+        {
+            index: 1233,
+            fields: {
+                height: '300',
+                width: '599',
+                isMaster: 'false',
+            },
+            mediaType: 'picture',
+            mimeType: 'jpg',
+            url: 'https://media.guim.co.uk/abc/599',
+        },
+        {
+            index: 1234,
+            fields: {
+                height: '600',
+                width: '1000',
+                isMaster: 'false',
+            },
+            mediaType: 'picture',
+            mimeType: 'jpg',
+            url: 'https://media.guim.co.uk/abc/1000',
+        },
+        {
+            index: 1235,
+            fields: {
+                height: '900',
+                width: '1500',
+                isMaster: 'false',
+            },
+            mediaType: 'picture',
+            mimeType: 'jpg',
+            url: 'https://media.guim.co.uk/abc/1500',
+        },
+    ];
+    const bestFitImage = getImage(images, 600);
+    if (bestFitImage) {
+        expect(bestFitImage.url).toBe('https://media.guim.co.uk/abc/1000');
+    }
+});
+
+test('returns null if no image', async () => {
```

Comment:
I'm not sure we want to silently return null in this case?

If this is just to be used for AMP, I think we want to throw an exception when we can't render. I think we decided this, but just haven't documented it. @nicl?

Reply:
okay cool. Can amend this